pkg/autodiscovery/providers/utils_test.go

```diff
@@ -8,45 +8,85 @@ package providers
 import (
 	"errors"
 	"fmt"
+	"reflect"
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-
 	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
 	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestParseJSONValue(t *testing.T) {
-	// empty value
-	res, err := parseJSONValue("")
-	assert.Nil(t, res)
-	assert.NotNil(t, err)
-
-	// value is not a list
-	res, err = parseJSONValue("{}")
-	assert.Nil(t, res)
-	assert.NotNil(t, err)
-
-	// invalid json
-	res, err = parseJSONValue("[{]")
-	assert.Nil(t, res)
-	assert.NotNil(t, err)
-
-	// bad type
-	res, err = parseJSONValue("[1, {\"test\": 1}, \"test\"]")
-	assert.Nil(t, res)
-	assert.NotNil(t, err)
-	assert.Equal(t, "found non JSON object type, value is: '1'", err.Error())
-
-	// valid input
-	res, err = parseJSONValue("[{\"test\": 1}, {\"test\": 2}]")
-	assert.Nil(t, err)
-	assert.NotNil(t, res)
-	require.Len(t, res, 2)
-	assert.Equal(t, integration.Data("{\"test\":1}"), res[0])
-	assert.Equal(t, integration.Data("{\"test\":2}"), res[1])
+	tests := []struct {
+		name          string
+		inputValue    string
+		want          [][]integration.Data
+		wantErr       bool
+		wantErrString string
+	}{
+		{
+			name:          "empty value",
+			inputValue:    "",
+			wantErr:       true,
+			wantErrString: "Value is empty",
+			want:          nil,
+		},
+		{
+			name:       "value is not a list",
+			inputValue: "{}",
+			wantErr:    true,
+			want:       nil,
+		},
+		{
+			name:       "invalid json",
+			inputValue: "[{]",
+			wantErr:    true,
+			want:       nil,
+		},
+		{
+			name:          "bad type",
+			inputValue:    "[1, {\"test\": 1}, \"test\"]",
+			wantErr:       true,
+			wantErrString: "failed to decode JSON Object '1' to integration.Data struct: found non JSON object type, value is: '1'",
+			want:          nil,
+		},
+		{
+			name:       "valid input",
+			inputValue: "[{\"test\": 1}, {\"test\": 2}]",
+			wantErr:    false,
+			want: [][]integration.Data{
+				{integration.Data("{\"test\":1}")},
+				{integration.Data("{\"test\":2}")},
+			},
+		},
+		{
+			name:       "valid input with list",
+			inputValue: "[[{\"test\": 1.1},{\"test\": 1.2}], {\"test\": 2}]",
+			wantErr:    false,
+			want: [][]integration.Data{
+				{integration.Data("{\"test\":1.1}"), integration.Data("{\"test\":1.2}")},
+				{integration.Data("{\"test\":2}")},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := parseJSONValue(tt.inputValue)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("parseJSONValue() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if tt.wantErr && tt.wantErrString != "" {
+				if err.Error() != tt.wantErrString {
+					t.Errorf("parseJSONValue() error = %v, wantErrString %v", err, tt.wantErrString)
+				}
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
`assert.Equal` already does a `DeepEqual` [under the hood](https://github.com/stretchr/testify/blob/master/assert/assertions.go#L56), could we use this one instead?

See https://github.com/DataDog/datadog-agent/blob/30be2879de237229f76b8754353b57cff7d32492/pkg/util/kubernetes/apiserver/endpoints_test.go#L81-L90 for a similar check flow